### PR TITLE
Added support for EV charger energy management.

### PR DESCRIPTION
**Pull request by rstraczynski**

### CODE REVIEW

File: README.md
Comment (boguslaw-wojcik) [inline README.md:973]: Please sync with Markus and remove smart_charge and rated_current if it is not used by any of our adapters (which I think it is not). We pollute the documentation with features that may never be supported, let's clean this up.

File: README.md
Comment (boguslaw-wojcik) [inline README.md:996]: sup_errors does not make sense from my point of view and it's probably not used. Please sync with Markus and clean this up if it's not used/required.

File: README.md
Comment (boguslaw-wojcik) [inline README.md:983]: Put the object example in this section instead of Examples at the end. See this example for reference:
image

File: README.md
Comment (boguslaw-wojcik) [inline README.md:959]: Let's give @bjhandeland no reasons to play a grammar tyrant here! :-) Add periods at the end of each sentence in the description column.

File: README.md
Comment (boguslaw-wojcik) [inline README.md:994]: Same with sup_access. If it was not used or it is not interpreted in any way by UI, let's remove it from spec. Double check with @GenixPL and Markus.

File: README.md
Comment (boguslaw-wojcik) [inline README.md:983]: Make it obvious it's about max values, like: Command for setting an electric current MAXIMUM amperage.

File: README.md
Comment (boguslaw-wojcik) [inline README.md:980]: ...for the current session.

File: README.md
Comment (rstraczynski) [reply README.md:973]: Markus confirmed it's not used anywhere, I'll remove it...

File: README.md
Comment (rstraczynski) [reply README.md:996]: Markus confirmed it's not used anywhere, I'll remove it...

File: README.md
Comment (rstraczynski) [reply README.md:994]: Markus confirmed it's not used anywhere inside edge apps. Waiting for confirmation from Łukasz...

File: README.md
Comment (rstraczynski) [reply README.md:973]: Done.

File: README.md
Comment (rstraczynski) [reply README.md:996]: Done.

File: README.md
Comment (rstraczynski) [reply README.md:994]: Confirmed from Łukasz's side, too. Removing it...

File: README.md
Comment (rstraczynski) [reply README.md:980]: Fixed.

File: README.md
Comment (rstraczynski) [reply README.md:994]: Fixed.

File: README.md
Comment (rstraczynski) [reply README.md:959]: Done.

File: README.md
Comment (rstraczynski) [reply README.md:983]: Fixed.

File: README.md
Comment (boguslaw-wojcik) [inline README.md:969]: Trash this too if you removed sup_access.

Comment (rstraczynski) [reply README.md]: Right, fixed.

**Review by boguslaw-wojcik — DISMISSED** [review]
Approval from my side. Please get approval from Christer as well.

File: README.md
Comment (bjhandeland) [inline README.md:982]: requesting is listed twice.

File: README.md
Comment (bjhandeland) [inline README.md:984]: sup_access and sup_errors removed intentionally?

File: README.md
Comment (bjhandeland) [inline README.md:983]: With a range it's usually a good idea to have something like `sup_steps` or `sup_increments` as well. Like, if the range is 7.0 to 40.0, does that mean I can set `7.1`? Or can I only set 7 or 8? Or maybe even only half, like 7, 7.5 and then 8?

File: README.md
Comment (bjhandeland) [inline README.md:981]: nit: not strictly necessary to make it this long 😅

**Review by bjhandeland — CHANGES_REQUESTED** [review]
some questions / comments

File: README.md
Comment (rstraczynski) [reply README.md]: Good catch, fixed!

File: README.md
Comment (rstraczynski) [reply README.md]: Outdated - requirements have changed already.

**Review by boguslaw-wojcik — CHANGES_REQUESTED** [review]
IMHO off is bad. Adapter should persist on its own charging mode and use it for every charging session, but off introduced whole new level of complications and consequences.
If the last mode was set to off and someone wants to start new session and clicks start - what will happen? Will you block charging for the user until he changes mode to slow or normal? Or will you pick at random slow or normal for the new session?

**Review by boguslaw-wojcik — APPROVED** [review]
That was fast...

**Review by boguslaw-wojcik — DISMISSED** [review]

File: README.md
Comment (boguslaw-wojcik) [inline README.md:974]: This needs to be split into "Interface props" and "Service props".

File: README.md
Comment (boguslaw-wojcik) [inline README.md:959]: charging mode in inverted `

Comment (rstraczynski) [reply README.md]: Done.

**Review by boguslaw-wojcik — DISMISSED** [review]

**Review by bjhandeland — APPROVED** [review]
lgtm

**Review by chribru — APPROVED** [review]

**Pull request merged**
